core/src/main/java/com/klaytn/caver/validator/Validator.java

```diff
@@ -0,0 +1,227 @@
+     * This function will compare public key in account key information from Klaytn and public key recovered form signature.
+     * <pre>Example :
+     * {@code
+     * String address = "0x{address}";
+     * String message = "Some data";
+     * List<SignatureData> sigDataList = Arrays.asList(new SignatureData("0x{v data}", "0x{r data}", "0x{s data}"), .....);
+     *
+     * boolean isValid = caver.validator.validateSignedMessage(message, sigDataList, address);
+     * }
+     * </pre>
+     *
+     * @param message The raw message string. If you want to pass data that already hashed with Klaytn specific prefix, see {@link Validator#validateSignedMessage(String, List, String, boolean)}
+     * @param signatures The list of  {@link SignatureData} instance to validate signatures.
+     * @param address The address of the account that signed the message.
+     * @param isHashed If the `isHashed` is true, the given message will NOT automatically be prefixed with "\x19Klaytn Signed Message:\n" + message.length + message, and be assumed as already prefixed.
+     * @return boolean
+     */
+    public boolean validateSignedMessage(String message, List<SignatureData> signatures, String address, boolean isHashed) {
+        try {
+            //Loading Account Key.
+            AccountKey accountKey = klay.getAccountKey(address).send();
+            if(accountKey.hasError()) {
+                throw new RuntimeException("error code : " + accountKey.getError().getCode() + " error message : " + accountKey.getError().getMessage());
+            }
+
+            //Extract public keys from signatures.
+            List<String> pubKeys = new ArrayList<>();
+            for(SignatureData signature : signatures) {
+                pubKeys.add(Utils.recoverPublicKey(message, signature, isHashed));
+            }
+
+            //Compare a account key from queried and public keys extracting from signature.
+            return validateWithAccountType(address, accountKey.getResult().getAccountKey(), pubKeys, AccountKeyRoleBased.RoleGroup.TRANSACTION.getIndex());
+        } catch(IOException e) {
+            throw new RuntimeException("Failed to get AccountKey from Klaytn", e);
+        } catch(SignatureException e) {
+            throw new RuntimeException("Failed to recover signature", e);
+        }
+    }
+
+     private boolean validateWithAccountType(String address, IAccountKey accountKey, List<String> pubKeys, int role) {
+        // For accounts that have not yet been applied in Klaytn's state, the return value of `caver.rpc.klay.getAccountKey` is null.
+        // In this case, the account's key has never been updated, so the logic is the same as in AccountKeyLegacy.
+        if(accountKey == null) {
+            accountKey = new AccountKeyLegacy();
+        }
+
+        Account account = new Account(address, accountKey);
+
+        if(accountKey instanceof AccountKeyLegacy) {
+            return validateAccountKeyLegacy(account, pubKeys.get(0));
```

Comment:
Please check whole signature array element to check whether include valid sig or not.